ext/web/16_image_data.js

```diff
@@ -0,0 +1,192 @@
+          "IndexSizeError",
+        );
+      }
+
+      if (data.length % 4 !== 0) {
+        throw new DOMException(
+          "Failed to construct 'ImageData': The input data length is not a multiple of 4.",
+          "InvalidStateError",
+        );
+      }
+
+      if (data.length / 4 % sourceWidth !== 0) {
+        throw new DOMException(
+          "Failed to construct 'ImageData': The input data length is not a multiple of (4 * width).",
+          "IndexSizeError",
+        );
+      }
+
+      if (
+        webidl.type(sourceHeight) !== "Undefined" &&
+        (sourceWidth * sourceHeight * 4 !== data.length)
+      ) {
+        throw new DOMException(
+          "Failed to construct 'ImageData': The input data length is not equal to (4 * width * height).",
+          "IndexSizeError",
+        );
+      }
+
+      if (webidl.type(sourceHeight) === "Undefined") {
+        this.#height = data.length / 4 / sourceWidth;
+      } else {
+        this.#height = sourceHeight;
+      }
+
+      if (
+        webidl.type(settings) === "Object" &&
+        webidl.type(settings.colorSpace) !== "Undefined"
+      ) {
+        this.#colorSpace = webidl.converters.PredefinedColorSpace(
+          settings.colorSpace,
+          prefix,
+          "colorSpace",
+        );
+      } else {
+        this.#colorSpace = "srgb";
+      }
+
+      this.#width = sourceWidth;
+      this.#data = data;
+      return;
+    }
+
+    // Overload: new ImageData(sw, sh [, settings])
+    sourceWidth = webidl.converters["unsigned long"](
+      arg0,
+      prefix,
+      "Argument 1",
+    );
+    sourceHeight = webidl.converters["unsigned long"](
+      arg1,
+      prefix,
+      "Argument 2",
+    );
+
+    if (sourceWidth < 1) {
+      throw new DOMException(
+        "Failed to construct 'ImageData': The source width is zero or not a number.",
+        "IndexSizeError",
+      );
+    }
+
+    if (sourceHeight < 1) {
+      throw new DOMException(
+        "Failed to construct 'ImageData': The source height is zero or not a number.",
+        "IndexSizeError",
+      );
+    }
+
+    if (
+      webidl.type(settings) === "Object" &&
+      webidl.type(settings.colorSpace) !== "Undefined"
+    ) {
+      this.#colorSpace = webidl.converters.PredefinedColorSpace(
+        settings.colorSpace,
+        prefix,
+        "colorSpace",
+      );
+    } else {
+      this.#colorSpace = "srgb";
+    }
+
+    this.#width = sourceWidth;
+    this.#height = sourceHeight;
+    this.#data = new Uint8ClampedArray(sourceWidth * sourceHeight * 4);
+  }
+
+  get width() {
+    webidl.assertBranded(this, ImageDataPrototype);
+    return this.#width;
+  }
+
+  get height() {
+    webidl.assertBranded(this, ImageDataPrototype);
+    return this.#height;
+  }
+
+  get data() {
+    webidl.assertBranded(this, ImageDataPrototype);
+    return this.#data;
+  }
+
+  get colorSpace() {
+    webidl.assertBranded(this, ImageDataPrototype);
+    return this.#colorSpace;
+  }
+}
```

Comment:
Please add `Deno.privateCustomInspect` method

Reply:
Thanks! I've implemented with `createFilteredInspectProxy` as it seems like a common pattern.

If there's a preference to use another approach, please let me know.

Reply:
Please pass `inspectOptions`

```js
[SymbolFor("Deno.privateCustomInspect")](inspect, inspectOptions) {
  return inspect(.., inspectOptions);
}
```